respx/models.py

```diff
@@ -235,19 +235,27 @@ def __init__(
         self.stats = mock.MagicMock()
 
     @property
-    def called(self):
+    def called(self) -> bool:
         return self.stats.called
 
     @property
-    def call_count(self):
+    def call_count(self) -> int:
         return self.stats.call_count
 
     @property
-    def calls(self):
+    def calls(self) -> List[Tuple[Request, ResponseTemplate]]:
         return [
             (request, response) for (request, response), _ in self.stats.call_args_list
         ]
 
+    @property
+    def last_request(self) -> Optional[Request]:
+        calls = self.calls
+        if not calls:
+            return None
+
+        return calls[-1][0]
```

Comment:
I think this feature could be implemented a bit different to not have such a specific property on the `RequestPattern` level.

Also, I'd expect same feature to be accessible through global `respx` api.

I've had the idea for a long time to break out the `calls` to a separat stats object with nested call objects, to get more flexibility for features like this.

I agree that it's currently a bit ugly  to get hold of a request by `request_pattern.calls[-1][0]`.

How about being able to do...
```py
request_pattern.calls[-1].request
```

...and also...

```py
request_pattern.calls.last.request
```

To not break backward compatibility, it would be nice to still be able to do...

```py
request, response = request_pattern.calls[-1]
# and/or
request, response = request_pattern.calls.last
```

...but that would mean that a call object needs to be a `tuple` but with request and response properties, if possible ;-)